sdk/timeseriesinsights/Azure.Iot.TimeSeriesInsights/design/ModelSettings_API_Design.md

```diff
@@ -0,0 +1,89 @@
+
+### GET /timeseries/modelSettings
+
+Returns the model settings which includes model display name, Time Series ID properties and default type ID. Every Gen2 environment has a model that is automatically created.
+
+```json
+{
+  "parameters": {
+    "api-version": "2020-07-31",
+    "environmentFqdn": "10000000-0000-0000-0000-100000000109.env.timeseries.azure.com"
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "modelSettings": {
+          "name": "DefaultModel",
+          "timeSeriesIdProperties": [
+            {
+              "name": "DeviceId",
+              "type": "String"
+            }
+          ],
+          "defaultTypeId": "5AB70D71-A8CD-410E-B70D-6F04AB9C132C"
+        }
+      }
+    }
+  }
+}
+```
+
+```csharp
+ /// <summary>
+/// Gets model settings asynchronously.
+/// </summary>
+/// <param name="clientSessionId">A client session ID allows the service to trace a group of related operations across services.</param>
+/// <param name="cancellationToken">The cancellation token.</param>
+/// <returns>The Model Settings which includes model display name, Time Series ID properties and default type ID with the http response <see cref="Response{T}"/>.</returns>
+public virtual async Task<Response<ModelSettingsResponse>> GetModelSettingsAsync(string clientSessionId = null, CancellationToken cancellationToken = default)
+```
+
+### POST /timeseries/modelSettings
+
+Updates time series model settings - either the model name or default type ID.
+
+```json
+{
+  "parameters": {
+    "api-version": "2020-07-31",
+    "environmentFqdn": "10000000-0000-0000-0000-100000000109.env.timeseries.azure.com",
+    "parameters": {
+      "name": "Thermostats"
+    }
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "modelSettings": {
+          "name": "Thermostats",
+          "timeSeriesIdProperties": [
+            {
+              "name": "DeviceId",
+              "type": "String"
+            }
+          ],
+          "defaultTypeId": "5AB70D71-A8CD-410E-B70D-6F04AB9C132C"
+        }
+      }
+    }
+  }
+}
+```
+```csharp
+ /// <summary>
+/// Updates model settings, either the model name or default type ID asynchronously.
+/// </summary>
+/// <param name="modelSettings">Model settings update request body.</param>
+/// <param name="clientSessionId">A client session ID allows the service to trace a group of related operations across services.</param>
+/// <param name="cancellationToken">The cancellation token.</param>
+/// <returns>The updated Model Settings with the http response <see cref="Response{T}"/>.</returns>
+/// <exception cref="ArgumentNullException"> <paramref name="modelSettings"/> is null. </exception>
+public virtual async Task<Response<ModelSettingsResponse>> UpdateModelSettingsAsync(UpdateModelSettingsRequest modelSettings, string clientSessionId = null, CancellationToken cancellationToken = default)
```

Comment:
Name suggestions for UpdateModelSettingsRequest customized class:
1. UpdateModelSettingsRequestOptions
2. UpdateModelSettings

Reply:
UpdateModelSettings is favored more and will be renaming to this.